raiden/blockchain/decode.py

```diff
@@ -410,7 +410,7 @@ def blockchainevent_to_statechange(
             proxy_manager=proxy_manager,
             chain_state=chain_state,
             event=event,
-            latest_confirmed_block=latest_confirmed_block,
+            current_confirmed_head=current_confirmed_head,
```

Comment:
This was the bug, the value here was the confirmed block as seen by the state machine, which could be old enough to be pruned. The fix is to use the confirmed head instead.